packages/client/src/schedule-helpers.ts

```diff
@@ -326,14 +343,9 @@ export async function decodeScheduleAction(
       args: await decodeArrayFromPayloads(dataConverter, pb.startWorkflow.input?.payloads),
       memo: await decodeMapFromPayloads(dataConverter, pb.startWorkflow.memo?.fields),
       retry: decompileRetryPolicy(pb.startWorkflow.retryPolicy),
-      searchAttributes: Object.fromEntries(
-        Object.entries(
-          mapFromPayloads(
-            searchAttributePayloadConverter,
-            pb.startWorkflow.searchAttributes?.indexedFields ?? {}
-          ) as SearchAttributes
-        )
-      ),
+      // TODO(thomas): is there a reason why we didn't filter out empty arrays here?
```

Comment:
IIRC Schedules were introduced in 1.18, at which point the server would no longer be sending empty arrays anyway.